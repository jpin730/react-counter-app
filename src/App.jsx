import PropTypes from "prop-types";

export const App = ({ title, subtitle }) => {
  return (
    <>
      <h1 data-testid="test-title" className="text-center m-5">
        {title}
      </h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
    </>
  );
};

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

// App.defaultProps = {
//   title: "Hello world!",
// };
