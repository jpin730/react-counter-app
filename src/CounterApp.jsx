import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);

  const handleAdd = () => setCounter(counter + 1);

  return (
    <div className="border rounded shadow m-5 p-5 text-center">
      <h1>CounterApp</h1>
      <h2 className="bg-success text-bg-success rounded w-25 py-3 mx-auto my-5">
        {counter}
      </h2>
      <button
        className="btn btn-danger btn-lg px-5 mx-3"
        onClick={handleSubtract}
      >
        -1
      </button>
      <button
        className="btn btn-secondary btn-lg px-5 mx-3"
        aria-label="btn-reset"
        onClick={handleReset}
      >
        Reset
      </button>
      <button className="btn btn-primary btn-lg px-5 mx-3" onClick={handleAdd}>
        +1
      </button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
