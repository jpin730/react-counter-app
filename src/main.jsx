import React from "react";
import ReactDOM from "react-dom/client";

import { App } from "./App";
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <App title="Hello world!" />
    <CounterApp value={10} />
  </React.StrictMode>
);
