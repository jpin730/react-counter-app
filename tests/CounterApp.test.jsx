import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("CounterApp", () => {
  const initialValue = 100;

  test("should match snapshot", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test("should show value as a h2", () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      initialValue.toString()
    );
  });

  test("should increase on click button '+1'", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));
    const newValueStr = (initialValue + 1).toString();
    expect(screen.getByText(newValueStr)).toBeTruthy();
  });

  test("should decrease on click button '-1'", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("-1"));
    const newValueStr = (initialValue - 1).toString();
    expect(screen.getByText(newValueStr)).toBeTruthy();
  });

  test("should reset on click button 'Reset'", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    const resetBtn = screen.getByRole("button", { name: "btn-reset" });
    fireEvent.click(resetBtn);
    expect(screen.getByText(initialValue.toString())).toBeTruthy();
  });
});
