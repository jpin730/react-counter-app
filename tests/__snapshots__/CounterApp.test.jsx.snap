// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should match snapshot 1`] = `
<div>
  <div
    class="border rounded shadow m-5 p-5 text-center"
  >
    <h1>
      CounterApp
    </h1>
    <h2
      class="bg-success text-bg-success rounded w-25 py-3 mx-auto my-5"
    >
      100
    </h2>
    <button
      class="btn btn-danger btn-lg px-5 mx-3"
    >
      -1
    </button>
    <button
      aria-label="btn-reset"
      class="btn btn-secondary btn-lg px-5 mx-3"
    >
      Reset
    </button>
    <button
      class="btn btn-primary btn-lg px-5 mx-3"
    >
      +1
    </button>
  </div>
</div>
`;
